### PR TITLE
Feature/webpack install

### PR DESCRIPTION
**Pull request by kjugi**
Adding the core of webpack files and structure to checkout.

#13

### CODE REVIEW

File: view/frontend/web/js/script.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import Vue from './lib/vue.js'
+import index from './components/index.vue'
+
+var app = new Vue({
+    el: '#checkout',
```
Comment (Koc) [inline view/frontend/web/js/script.js]: Is it possible to remove hardcode on element id from js files and determinate target element which was specified inside `x-magento-init`?

**Comment by Koc** [comment]
Looks really awesome! Thank you for work in this direction.

Some questions:
- how can we pass some Magento js libs as rependencies? For example if my Vue component requires `'Magento_Customer/js/customer-data'` - how can I require it using Webpack?
- how runtime config like `selectedMethods` passed to the index.vue?

File: view/frontend/web/js/script.js (hunk 1 above)
Comment (kjugi) [reply view/frontend/web/js/script.js]: That would be good

Comment (Koc) [reply view/frontend/web/js/script.js]: http://devdocs.magento.com/guides/v2.1/javascript-dev-guide/javascript/js_init.html

We can get dom element and passed config.

Comment (Igloczek) [reply view/frontend/web/js/script.js]: We are not going to support any core Magento features related to JS modules, b/c that's why we are writing it using Vue and Webpack, to make this simpler for front-ends, not follow the same overcomplicated practices from the core.

**Comment by Igloczek** [comment]
@Koc
1. You can't. It's a fully custom implementation and we are not going to use any existing JS code.
2. It will be stored in Vuex, for now we are using global variables.

**Comment by Koc** [comment]
have you plan to rewrite all frontend, like cart, swatch renderers, gallery, ...?

File: view/frontend/web/js/components/button.vue [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    {{ text }}
+  </button>
+</template>
+
+
+<script>
+export default {
+  props: {
+    buttonclass: {
+        type: String
```
Comment (kjugi) [inline view/frontend/web/js/components/button.vue]: add require elements to the necessary things

**Pull request merged**